spec.bs

```diff
@@ -347,6 +530,196 @@ Moreover, each {{SharedStorageWorklet}}'s [=global scopes|list of global scopes=
+
+  <div algorithm>
+    To <dfn>validate reporting metadata</dfn>, given an {{/object}} |reportingMetadata|, run the following steps:
+
+    1. If |reportingMetadata| is [=map/empty=], return true.
+    1. If |reportingMetadata| is not a [=dictionary=], return false.
+    1. [=map/iterate|For each=] <var ignore="">eventType</var> -> |urlString| of |reportingMetadata|, if the result of running [=get the canonical URL string if valid=] with |urlString| is undefined, return false.
+    1. Return true.
+  </div>
+
+  <div algorithm>
+    To <dfn>get the canonical URL string if valid</dfn>, given a [=string=] |urlString|, run the following steps:
+
+    1. Let |url| be the result of running a [=URL parser=] on |urlString|.
+    1. If |url| is not a valid [=/URL=], return undefined.
+    1. Otherwise, return the result of running a [=URL serializer=] on |url|.
+  </div>
+
+  <div algorithm>
+    To <dfn>register reporting metadata</dfn>, given an {{/object}} |reportingMetadata| and a "fenced frame config struct" |fencedFrameConfigStruct|, run the following steps:
+
+    1. If |reportingMetadata| is [=map/empty=], return.
+    1. [=Assert=]: |reportingMetadata| is a [=dictionary=].
+    1. Let |reportingUrlMap| be an [=map/empty=] [=map=].
+    1. [=map/iterate|For each=] |eventType| -> |urlString| of |reportingMetadata|:
+        1. Let |url| be the result of running a [=URL parser=] on |urlString|.
+        1. [=Assert=]: |url| is a valid [=/URL=].
+        1. [=map/Set=] |reportingUrlMap|[|eventType|] to |url|.
+    1. <span class=todo>Store |reportingUrlMap| inside a "fenced frame reporter" class associated with |fencedFrameConfigStruct|. Both of these still need to be added to the draft [=fenced frame|Fenced Frame specification=].</span>
+  </div>
+
+  ## Entropy Budgets ## {#budgets}
+
+  Because [=bits of entropy=] can leak via {{SharedStorageWorklet/selectURL()}}, the [=user agent=] will need to maintain budgets to limit these leaks.
+
+  ### Navigation Entropy Budget ### {#nav-budget}
+
+  If a user activates a [=fenced frame=] whose {{FencedFrameConfig}} was generated by {{SharedStorageWorklet/selectURL()}} and thereby initiates a [=top-frame=] [=navigate|navigation=], this will reveal to the landing page that its [=/URL=] was selected, which is a leak in [=entropy bits=] of up to logarithm base 2 of the number of input [=/URLs=] for the call to {{SharedStorageWorklet/selectURL()}}. To mitigate this, a [=user agent=] will set a per-[=calling site=] [=navigation entropy allowance=].
```

Comment:
It's unfortunately hard to suggest specific changes to massive massive blocks like this since they are not wrapped at all :(

But you should link "activates" to HTML, like we do in fenced frame: https://github.com/WICG/fenced-frame/blob/master/spec.bs#L167